lib/filterx/object-primitive.c

```diff
@@ -63,6 +63,26 @@ _integer_map_to_json(FilterXObject *s, struct json_object **object, FilterXObjec
   return TRUE;
 }
 
+static FilterXObject *
+_integer_add(FilterXObject *lhs_object, FilterXObject *rhs_object)
+{
+  if (!filterx_object_is_type(lhs_object, &FILTERX_TYPE_NAME(integer)))
+    goto error;
+  GenericNumber base = filterx_primitive_get_value(lhs_object);
+  if (filterx_object_is_type(rhs_object, &FILTERX_TYPE_NAME(integer)))
+    {
+      GenericNumber add = filterx_primitive_get_value(rhs_object);
+      return filterx_integer_new(gn_as_int64(&base) + gn_as_int64(&add));
+    }
+  else if (filterx_object_is_type(rhs_object, &FILTERX_TYPE_NAME(double)))
```

Comment:
[nitpick]

This `else if` could be a simple `if`, as you did an early return above.